PWGHF/HFL/HFmAnalysis.cxx

```diff
@@ -0,0 +1,341 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file HFmAnalysis.cxx
+/// \brief HF single muon analysis task
+/// \This task is derived from the DQ framework and it is used to extract the observables on single muons needed for the HF-muon analysis.
+/// \author Maolin Zhang <maolin.zhang@cern.ch>, CCNU
+
+#include "Framework/runDataProcessing.h"
+#include "Framework/AnalysisTask.h"
+#include "Framework/ASoAHelpers.h"
+#include "Framework/AnalysisDataModel.h"
+#include "PWGDQ/Core/VarManager.h"
+#include "PWGDQ/Core/HistogramManager.h"
+#include "PWGDQ/Core/AnalysisCut.h"
+#include "PWGDQ/Core/AnalysisCompositeCut.h"
+#include "PWGDQ/Core/CutsLibrary.h"
+#include "PWGDQ/Core/HistogramsLibrary.h"
+#include "Framework/HistogramRegistry.h"
+#include "PWGDQ/DataModel/ReducedInfoTables.h"
+#include "ReconstructionDataFormats/TrackFwd.h"
+#include "Common/DataModel/EventSelection.h"
+
+using namespace o2;
+using namespace o2::aod;
+using namespace o2::framework;
+
+namespace o2::aod
+{
+namespace dq_analysis_flags
+{
+DECLARE_SOA_COLUMN(IsEventSelected, isEventSelected, int);
+}
+
+DECLARE_SOA_TABLE(EventCuts, "AOD", "EVENTCUTS", dq_analysis_flags::IsEventSelected);
+} // namespace o2::aod
+
+using MyCollisions = o2::soa::Join<aod::Collisions, aod::EvSels>;
+using MyMcCollisions = o2::soa::Join<aod::Collisions, aod::McCollisionLabels, aod::EvSels>;
+using MyEventsSelected = o2::soa::Join<aod::Collisions, aod::EvSels, o2::aod::EventCuts>;
+using MyMcEventsSelected = o2::soa::Join<aod::Collisions, aod::McCollisionLabels, aod::EvSels, o2::aod::EventCuts>;
+using MyMuons = o2::soa::Join<aod::FullFwdTracks, aod::McFwdTrackLabels>;
+
+constexpr static uint32_t gMuonFillMap(VarManager::ObjTypes::Muon | VarManager::ObjTypes::MuonCov);
+constexpr static uint32_t gEventFillMap(VarManager::ObjTypes::BC | VarManager::ObjTypes::Collision);
+constexpr static uint32_t gTrackMCFillMap(VarManager::ObjTypes::ParticleMC);
+
+void DefineHistograms(HistogramManager*, TString);
+
+struct HFEventSelection {
```

Comment:
This is still too ambiguous. There are other event selections in other HF code. I would suggest:
```suggestion
struct HfmEventSelection {
```